services/course-material/src/components/ContentRenderer/ParagraphBlock/index.tsx

```diff
@@ -69,44 +71,101 @@ const ParagraphBlock: React.FC<BlockRendererProps<ParagraphAttributes>> = ({
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const memoizedContent = useMemo(() => editedContent, [selectedBlockId])
 
   if (editing) {
-    return (
-      <p
-        className={css`
-          ${normalWidthCenteredComponentStyles}
-          white-space: pre-line;
-          min-width: 1px;
-          color: ${textColor};
-          background-color: ${backgroundColor};
-          font-size: ${fontSize};
-          ${backgroundColor && `padding: 1.25em 2.375em;`}
-        `}
-        contentEditable
-        onInput={(ev) => {
-          const changed = ev.currentTarget.innerText
-          if (content !== changed) {
-            setEdits((edits) => {
-              edits.set(id, {
-                block_id: id,
-                block_attribute: "content",
-                original_text: content,
-                changed_text: changed,
+    if (selectedBlockId === id) {
+      // block focused, editing
+      return (
+        <p
+          className={css`
+            ${normalWidthCenteredComponentStyles}
+            white-space: pre-line;
+            min-width: 1px;
+            color: ${textColor};
+            background-color: ${backgroundColor};
+            font-size: ${fontSize};
+            ${backgroundColor && `padding: 1.25em 2.375em;`}
+            border: 1px;
+            border-style: dotted;
+          `}
+          contentEditable
+          onInput={(ev) => {
+            const changed = ev.currentTarget.innerText
+            setEditedContent(changed)
+            if (content !== changed) {
+              setEdits((edits) => {
+                edits.set(id, {
+                  block_id: id,
+                  block_attribute: "content",
+                  original_text: content,
+                  changed_text: changed,
+                })
+                return new Map(edits)
+              })
+            } else {
+              // returned to original
+              setEdits((edits) => {
+                edits.delete(id)
+                return new Map(edits)
               })
-              return new Map(edits)
-            })
-          } else {
-            // returned to original
-            setEdits((edits) => {
-              edits.delete(id)
-              return new Map(edits)
-            })
-          }
-        }}
-      >
-        {content}
-      </p>
-    )
+            }
+          }}
+        >
+          {memoizedContent}
+        </p>
+      )
+    } else {
+      const spans: JSX.Element[] = []
+      for (const diff of diffChars(data.attributes.content, editedContent)) {
+        // the diff spans should have the block id so the click handler can find it
+        if (diff.added) {
+          spans.push(
+            <span
+              className={css`
+                background: LightGreen;
+              `}
+            >
+              {diff.value}
+            </span>,
+          )
+        } else if (diff.removed) {
+          spans.push(
+            <span
```

Comment:
mark html element with aria label/description that says something like "removed text"